### PR TITLE
06 Readme docs

### PR DESCRIPTION
**Pull request by enoonan**

### CODE REVIEW

File: examples/06-custom-effects/README.md [hunk 1]
```diff
@@ -1,24 +1,73 @@
...
+
 # 06 Custom Effects
 
-We haven't quite got round to documenting this example yet. If you know a little
-bit about Lustre or Elm and want to help out, we'd love to have your help! Please
-[open an issue](https://github.com/lustre-labs/lustre/issues/new) if you have any
-ideas or reach out to @hayleigh.dev on the [Gleam discord](https://discord.gg/Fm8Pwmy).
+In the last example, we showed how to use effects provided by `lustre_http`. In
+this example we'll see how to create effects of our own using Lustre's
+[`effect.from`](https://hexdocs.pm/lustre/4.0.0-rc1/lustre/effect.html#from)
```
Comment (hayleigh-dot-dev) [inline examples/06-custom-effects/README.md:7]: We can link to the live docs now: https://hexdocs.pm/lustre/lustre/effect.html#from!

File: examples/06-custom-effects/README.md [hunk 2]
```diff
@@ -1,24 +1,73 @@
...
+method.
```
Comment (hayleigh-dot-dev) [inline examples/06-custom-effects/README.md:8]: Can we call them _functions_ pleased, Gleam isn't OOP we don't have methods.

File: examples/06-custom-effects/README.md [hunk 3]
```diff
@@ -1,24 +1,74 @@
...
+
+Since we use Lustre effects to communicate with _external_ systems (like the
```
Comment (hayleigh-dot-dev) [inline examples/06-custom-effects/README.md:10]: I think we should consistently refer to this as just "effects" or `Effect` (singular) – I think people know we're talking about Lustre's stuff at this point ^.^

File: examples/06-custom-effects/README.md [hunk 4]
```diff
@@ -1,24 +1,74 @@
...
+be sure to read up on that!
+
+> Gleam externals are part of its "foreign function interface", which is why
+> you'll typically see files with `ffi` in the name - like
+> [`app.ffi.mjs`](./src/app.ffi.mjs).
+
+## Accessing Browser Storage
+
+In this example, the external system we want to interact with is the browser's
+local storage. This way, we can write a message into the text input and it will
```
Comment (hayleigh-dot-dev) [inline examples/06-custom-effects/README.md:23]: Let's link out to the `localStorage` docs as a courtesy: https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage

File: examples/06-custom-effects/README.md [hunk 5]
```diff
@@ -1,24 +1,74 @@
...
+still be there when we refresh the page. Handy!
+
+The `view`, `update` and `init` functions should look pretty familiar by now, so
+let's focus on the functions that generate our custom effects.
+
+````rust fn read_localstorage(key: String) -> Effect(Msg) {
+    effect.from(fn(dispatch) {
+	do_read_localstorage(key) |> CacheUpdatedMessage |> dispatch
```
Comment (hayleigh-dot-dev) [inline examples/06-custom-effects/README.md:32]: Can we make sure all the code snippets are formatted as Gleam's formatter would do it.

File: examples/06-custom-effects/README.md [hunk 6]
```diff
@@ -1,24 +1,77 @@
...
+    effect.from(fn(_) {
+        do_write_localstorage(key, value)
+    })
+}
+```
+
+Here, our effect is simpler. We tell the gleam compiler we don't need to use the
+`dispatch` function by replacing it with a [discard
+pattern](https://tour.gleam.run/everything/#basics-discard-patterns). Then we
+write to local storage, and no more work needs to be done.
```
Comment (hayleigh-dot-dev) [inline examples/06-custom-effects/README.md:57]: Maybe we could add another paragraph explaining why you'd still want to use an effect here even though we're not making use of the `dispatch`. Moving side effects out our `update` function makes it easier to test and easier to port to a [server component](https://hexdocs.pm/lustre/lustre/server_component.html) in the future.

**Review by hayleigh-dot-dev — CHANGES_REQUESTED** [review]

File: examples/06-custom-effects/README.md (hunk 6 above)
Comment (enoonan) [reply examples/06-custom-effects/README.md:59]: Updated! Does this work?

Comment (hayleigh-dot-dev) [reply examples/06-custom-effects/README.md:59]: Stunning, thank you!

**Pull request merged**